test/unit/yesno.spec.ts

```diff
@@ -214,6 +216,22 @@ describe('Yesno', () => {
         /Request does not match mock. Expected request #0 "POST http:\/\/example.com:80" to have path "\/my\/path", not "\/my\/foobar"/,
       );
     });
+
+    it('should accept an optional comparatorFn, which can throw to reject a mock', async () => {
+      const mockErrorMessage = 'some-error';
+      const mockError = new Error(mockErrorMessage);
+
+      const comparatorFn: ComparatorFn = (
+        intercepted: ISerializedRequest,
+        mock: ISerializedRequest,
+        metadata: IComparatorMetadata,
+      ): boolean => {
+        throw mockError;
+      };
+
+      yesno.mock([createMock()], { comparatorFn });
+      await expect(requestTestServer()).rejectedWith(mockErrorMessage);
```

Comment:
Can you add a test statement asserting we can pass with the custom comparator as well?

Reply:
@ianwsperber Excellent suggestion, I have added the additional test case 👍